src/Kernel/Slot.class.st

```diff
@@ -300,6 +300,11 @@ Slot >> readInContext: aContext [
 	^self read: aContext receiver
 ]
 
+{ #category : #accessing }
+Slot >> scope [
+	^ definingClass
```

Comment:
This may be not correct. But I did not check users of #scope.  
Is #scope about a visibility of a variable? 
If true then I think it should be #owningClass because the visibility of the variable inherited from a trait is a target class. 
From the other side I am not sure (never used it) does method of a class with stateful trait have an access to a trait variables?

Reply:
Yes, it is #owningClass... the class where the slot is in that has the offsets correct to read the slot (the slot in the Trait knows nothing about the class...)

I will put it on the backlog, I want to write tests for alll the cases (this is not used so we can merge nevertheless)